activesupport/lib/active_support/callbacks.rb

```diff
@@ -331,23 +332,31 @@ def duplicates?(other)
 
         # Wraps code with filter
         def apply(callback_sequence)
-          user_conditions = conditions_lambdas
-          user_callback = CallTemplate.build(@filter, self)
-
-          case kind
-          when :before
-            Filters::Before.build(callback_sequence, user_callback.make_lambda, user_conditions, chain_config, @filter)
-          when :after
-            Filters::After.build(callback_sequence, user_callback.make_lambda, user_conditions, chain_config)
-          when :around
-            callback_sequence.around(user_callback, user_conditions)
-          end
+          public_send(kind, callback_sequence)
```

Comment:
What purpose is this change? It is hard to understand new methods are only called via the `public_send` to me.
And we generally do not accept such like cosmetic changes.
https://edgeguides.rubyonrails.org/contributing_to_ruby_on_rails.html#write-your-code

Reply:
Most changes in the PR are suggestions that provide small performance improvements. If you believe that using send makes the code too obscure, we can drop this commit.

Reply:
Yeah, this one I find a little dubious. Particularly because I remember @matthewd tuning the callback code to generate fewer entries in a backtrace, so we can't break that either (depending on the amount of stack pushes to performance gain ratio).

Reply:
No worries. Again, I made some of suggestions that made the code incrementally faster in the benchmark, but I expected some to be dropped.

Reply:
Note that extra method call doesn't make code faster in Ruby in my practice.
See also #36049, 89b86640901be4f423cef4059d970c0aa0adb648.